awx_collection/plugins/modules/tower_inventory_source.py

```diff
@@ -20,333 +20,346 @@
-        source=dict(required=True,
-                    choices=SOURCE_CHOICES.keys()),
-        credential=dict(required=False),
-        source_vars=dict(required=False),
-        timeout=dict(type='int', required=False),
-        source_project=dict(required=False),
-        source_path=dict(required=False),
-        update_on_project_update=dict(type='bool', required=False),
-        source_regions=dict(required=False),
-        instance_filters=dict(required=False),
-        group_by=dict(required=False),
-        source_script=dict(required=False),
-        overwrite=dict(type='bool', required=False),
-        overwrite_vars=dict(type='bool', required=False),
-        custom_virtualenv=dict(type='str', required=False),
-        update_on_launch=dict(type='bool', required=False),
-        update_cache_timeout=dict(type='int', required=False),
-        organization=dict(type='str'),
+        #
+        # How do we handle manual and file? Tower does not seem to be able to activate them
+        #
+        source=dict(choices=["manual", "file", "scm", "ec2", "gce",
+                             "azure_rm", "vmware", "satellite6", "cloudforms",
+                             "openstack", "rhv", "tower", "custom"], required=False),
+        source_path=dict(),
+        source_script=dict(),
+        source_vars=dict(type='dict'),
+        credential=dict(),
+        source_regions=dict(),
+        instance_filters=dict(),
+        group_by=dict(),
+        overwrite=dict(type='bool'),
+        overwrite_vars=dict(type='bool'),
+        custom_virtualenv=dict(type='str'),
+        timeout=dict(type='int'),
+        verbosity=dict(type='int', choices=[0, 1, 2]),
+        update_on_launch=dict(type='bool'),
+        update_cache_timeout=dict(type='int'),
+        source_project=dict(type='str'),
+        update_on_project_update=dict(type='bool'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
-    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
-
+    # One question here is do we want to end up supporting this within the ansible module itself (i.e. required if, etc)
+    # Or do we want to let the API return issues with "this dosen't support that", etc.
+    #
+    # GUI OPTIONS:
+    # - - - - - - - manual:	file:	scm:	ec2:	gce	azure_rm	vmware	sat	cloudforms	openstack	rhv	tower	custom
+    # credential		?	?	o	o	r	r		r	r	r		r		r	r	o
+    # source_project		?	?	r	-	-	-		-	-	-		-		-	-	-
+    # source_path		?	?	r	-	-	-		-	-	-		-		-	-	-
+    # verbosity			?	?	o	o	o	o		o	o	o		o		o	o	o
+    # overwrite			?	?	o	o	o	o		o	o	o		o		o	o	o
+    # overwrite_vars		?	?	o	o	o	o		o	o	o		o		o	o	o
+    # update_on_launch		?	?	o	o	o	o		o	o	o		o		o	o	o
+    # update_on_project_launch	?	?	o	-	-	-		-	-	-		-		-	-	-
+    # source_regions		?	?	-	o	o	o		-	-	-		-		-	-	-
+    # instance_filters		?	?	-	o	-	-		o	-	-		-		-	o	-
+    # group_by			?	?	-	o	-	-		o	-	-		-		-	-	-
+    # source_vars*		?	?	-	o	-	o		o	o	o		o		-	-	-
```

Comment:
If regions, filters, group_by, or source_vars are accepted by the API, but not functional for that particular inventory source type, then let's file it as an API bug.

Reply:
We have a task to file a bug for the inventory_sources in general. Try doing a POST to api/v2/inventory_sources with the payload:
```
{
"name": "Some Fubar Source",
"source": "scm",
"inventory": 1
}
```

 spolier alert:
![image](https://user-images.githubusercontent.com/32551173/74346512-b1f38a00-4d7d-11ea-8a67-5d9f3c72eb94.png)

Based on the epic state of this failure I didn't even bother trying all of the various other combinations and just put a little elbow grease into the module to try and enforce proper state.